src/cdomains/intDomain.ml

```diff
@@ -349,6 +349,446 @@ struct
+        else if b = y1  then meet_lists ~acc:((a, b)::acc) xs1 ((x2, y2)::xs2)
+        else                 meet_lists ~acc:((a, b)::acc) ((x1, y1)::xs1) xs2
+    | _, [] | [], _ ->
+      if acc = [] then None else Some (SS.of_list acc)
+
+  (* recursively comparing lists (they are assumed to be sorted) *)
+  let rec compare_lists l1 l2 =
+    match l1, l2 with
+    | (x1, x2) :: xs, (y1, y2) :: ys ->
+      if Int64.compare x1 y1 = 0 && Int64.compare x2 y2 = 0 then
+        compare_lists xs ys
+      else false
+    | [], [] -> true
+    | _, _ -> false
+
+  (* since s1 and s2 are basically unions of intervals, s1 <= s2 iff s1 meet s2 = s1 *)
+  let leq s1 s2 =
+    match s1, s2 with
+    | None, _ -> true
+    | Some _, None -> false
+    | Some x, Some y ->
+      let l1 = List.sort IntPair.compare (SS.elements x) in
+      let l2 = List.sort IntPair.compare (SS.elements y) in
+      let s_meet = meet_lists l1 l2 in
+      match s_meet with
+      | None -> false
+      | Some s -> compare_lists (List.sort IntPair.compare (SS.elements s)) l1
+
+  let equal x y =
+    match x, y with
+    | None, None -> true
+    | Some s1, Some s2 ->
+      SS.equal s1 s2
+    | _ -> false
+
+  let norm = function None -> None | Some (x,y) ->
+    if Int64.compare x y > 0 then None
+    else if Int64.compare min_int x > 0 || Int64.compare max_int y < 0 then top ()
+    else Some (SS.add (x,y) SS.empty)
+
+  let norm_interval (x, y) =
+    if Int64.compare min_int x > 0 || Int64.compare max_int y < 0 then (min_int, max_int)
+    else (x, y)
+
+  let (@@) f x = f x
+
+  let to_int = function
+    | Some s when SS.cardinal s = 1 ->
+        let (x, y) = SS.min_elt s in
+        if Int64.compare x y = 0 then Some x else None
+    | _ -> None
+
+  let of_int x =
+    let s = SS.empty in
+    let s_of_int = SS.add (x, x) s in
+    Some s_of_int
+
+  let is_int = function
+    | Some s when SS.cardinal s = 1 ->
+        let x, y = SS.min_elt s in
+        if Int64.compare x y = 0 then true else false
+    | _ -> false
+
+  let equal_to i = function
+    | None -> failwith "unsupported: equal_to with bottom"
+    | Some s ->
+      if not (SS.exists (fun (a, b) -> a <= i && i <= b) s) then `Neq else
+        match SS.cardinal s with
+          | 1 ->
+            let a, b = SS.min_elt s in
+            if a = b && b = i then `Eq else `Top
+          | _ -> `Top
+
+  let to_bool = function
+    | None -> None
+    | x when equal x zero -> Some false
+    | x -> if leq zero x then None else Some true
+
+  let to_bool_interval x =
+    match x with
+    | None -> None
+    | _ when equal x zero -> x
+    | _ -> if leq zero x then top_bool else one
+
+  let of_bool = function
+    | true -> Some (SS.add (1L, 1L) SS.empty)
+    | false -> Some (SS.add (0L, 0L) SS.empty)
+
+  let is_bool x =
+    x <> None && not (leq zero x) || equal x zero
+
+  let to_excl_list _ = None
+  let of_excl_list t _ = top ()
```

Comment:
You should be able to do better here!
An exclusion set with `n` excluded values should give rise to an interval set with maximum `n+1` intervals, right?